scripts/concurrent_task_utils.py

```diff
@@ -109,39 +111,37 @@ def __init__(
         self.semaphore = semaphore
         self.finished = False
         self.report_enabled = report_enabled
+        self.errors_to_retry_on = errors_to_retry_on or []
+        self.num_attempts = 0
 
     def run(self) -> None:
+        """Executes the task and retries on specified errors."""
         try:
-            self.task_results = self.func()
-            if self.verbose:
-                for task_result in self.task_results:
-                    # The following section will print the output of the lint
-                    # checks.
-                    if self.report_enabled:
+            while self.num_attempts < MAX_ATTEMPTS:
+                self.num_attempts += 1
+                try:
+                    self.task_results = self.func()
+                    log(
+                        f'FINISHED {self.name}: '
+                        f'{time.time() - self.start_time:.1f} secs',
+                        show_time=True
+                    )
+                    return
+                except Exception as e:
+                    # Check if the error is in the list of retryable errors.
+                    if any(err in str(e) for err in self.errors_to_retry_on):
                         log(
-                            'Report from %s check\n'
-                            '----------------------------------------\n'
-                            '%s' % (task_result.name, '\n'.join(
-                                task_result.get_report())), show_time=True)
-                    # The following section will print the output of backend
-                    # tests.
-                    else:
-                        log(
-                            'LOG %s:\n%s'
-                            '----------------------------------------' %
-                            (self.name, task_result.messages[0]),
-                            show_time=True)
-            log(
-                'FINISHED %s: %.1f secs' % (
-                    self.name, time.time() - self.start_time), show_time=True)
-        except Exception as e:
-            self.exception = e
-            self.stacktrace = traceback.format_exc()
-            if 'KeyboardInterrupt' not in self.exception.args[0]:
```

Comment:
What happened to this logic?

In general, for any old code you delete, please justify the deletion with a comment on the GitHub UI so that reviewers understand the rationale for the deletion.